pkg/operator/controller/ingress/controller.go

```diff
@@ -1205,3 +1220,42 @@ func (r *reconciler) allRouterPodsDeleted(ingress *operatorv1.IngressController)
 
 	return true, nil
 }
+
+// updateInfraIngressLoadBalancerIPs updates GCP's PlatformStatus with Ingress LB IPs when the DNSType is `ClusterHosted`.
+func updateInfraIngressLoadBalancerIPs(service *corev1.Service, infraConfig *configv1.Infrastructure) (bool, error) {
+	platformStatus := infraConfig.Status.PlatformStatus
+	if platformStatus != nil {
+		switch platformStatus.Type {
+		case configv1.GCPPlatformType:
+			if platformStatus.GCP != nil && platformStatus.GCP.CloudLoadBalancerConfig != nil && platformStatus.GCP.CloudLoadBalancerConfig.DNSType == configv1.ClusterHostedDNSType {
+				// The cluster has to run its own CoreDNS pod for DNS. Update Infra CR
+				// with the Ingress LB IPs. These values are used to configure the
+				// in-cluster DNS to provide resolution for *.apps.
+				if infraConfig.Status.PlatformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted == nil {
+					infraConfig.Status.PlatformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted = &configv1.CloudLoadBalancerIPs{}
```

Comment:
Any reason not to use `platformStatus` status here?
```suggestion
				if platformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted == nil {
					platformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted = &configv1.CloudLoadBalancerIPs{}
```
This makes the lines a little shorter, but more importantly, it makes it more obvious that you aren't potentially dereferencing a nil pointer since you checked `platformStatus.GCP != nil && platformStatus.GCP.CloudLoadBalancerConfig` above.